lib/collection.js

```diff
@@ -81,8 +81,9 @@ Collection.prototype.insert = function (docOrDocs, cb) {
     for (var i = 0; i < docs.length; i++) {
       if (!docs[i]._id) docs[i]._id = oid()
     }
-    server.insert(self._fullColName(), docs, writeOpts, function (err) {
+    server.insert(self._fullColName(), docs, writeOpts, function (err, res) {
       if (err) return cb(err)
+      if (res.result.writeErrors && res.result.writeErrors.length > 0) return cb(res.result.writeErrors[0])
```

Comment:
I just took a quick look through the mongodb-core code.  Every call I see returns either an error or a result.  The actual final success call happens at [line 951 of server.js](https://github.com/christkv/mongodb-core/blob/master/lib/topologies/server.js#L951).

But, I haven't worked with the core before (or anything other than mongojs for that matter) and I can assume from your comment that there are cases where no result is returned, so I'll update the code.